lib/web_ui/test/engine/view_embedder/flutter_view_manager_test.dart

```diff
@@ -0,0 +1,99 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:js_util';
+
+import 'package:test/bootstrap/browser.dart';
+import 'package:test/test.dart';
+import 'package:ui/src/engine.dart';
+
+import '../../common/matchers.dart';
+import '../../common/test_initialization.dart';
+
+void main() {
+  internalBootstrapBrowserTest(() => doTests);
+}
+
+Future<void> doTests() async {
+  await bootstrapAndRunApp();
+
+  group('FlutterViewManager', () {
+    int nextViewId = 1; // We keep track of this so we don't have to unregister from PlatformDispatcher.
+
+    final EnginePlatformDispatcher platformDispatcher = EnginePlatformDispatcher.instance;
+    final FlutterViewManager viewManager = FlutterViewManager();
+
+    group('registerView', () {
+      test('can register view', () {
+        final int viewId = nextViewId++;
+        final EngineFlutterView view = EngineFlutterView(viewId, platformDispatcher, createDomElement('div'));
+
+        viewManager.registerView(view);
+
+        expect(viewManager[viewId], view);
+      });
+
+      test('fails if the same viewId is already registered', () {
+        final int viewId = nextViewId++;
+        final EngineFlutterView view = EngineFlutterView(viewId, platformDispatcher, createDomElement('div'));
+
+        viewManager.registerView(view);
+
+        expect(() { viewManager.registerView(view); }, throwsAssertionError);
+      });
```

Comment:
...this!

Reply:
(The assertion thrown when registering duplicated views was breaking the `platform_dispatcher_test.dart` that does both create views, and then registers them)

Reply:
I actually had to allow overwriting of the ImplicitView for some tests to continue working as expected, like `test/engine/routing_test.dart`!

Reply:
Let's land this as-is for now. But I'm probably going to change it. Tests that need to override the implicit view can simply unregister the old one before registering the new one.